src/main/scala/viper/gobra/frontend/info/implementation/property/Assignability.scala

```diff
@@ -108,7 +109,7 @@ trait Assignability extends BaseProperty { this: TypeInfoImpl =>
 
   lazy val assignable: Property[PExpression] = createBinaryProperty("assignable") {
     case PIndexedExp(b, _) => underlyingType(exprType(b)) match {
-      case _: ArrayT => assignable(b)
+      case _: ArrayT => assignable(b) && isMutableArray(b)
```

Comment:
Do you catch everything with that. I would have expected some case in the beginning like `case e _ if !isMutable(e) => ...`